### PR TITLE
mailman-web: use default Nixpkgs Django

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

The upstream version bound of <2.2 is overly pessimistic, and
mailman-web works just fine on Nixpkgs' default Django 2.2.  I've
applied a patch to relax the bound, which I have also sent
upstream[1] (which is why I included the patch rather than just doing
another sed).

[1]: https://gitlab.com/mailman/mailman-web/-/merge_requests/2

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by lsix** [comment]
This looks good to me, but I have not tested mailman-web with django-2.2.

@peti do you have a setup where you could give this a try?

**Comment by Mic92** [comment]
I think @benley has one.

**Comment by alyssais** [comment]
FWIW mine is running happily at https://spectrum-os.org/lists/archives with this PR.

I did have some trouble where it failed to migrate the database to the new Django version, but I think this was due to an unrelated issue that had resulted in me having a corrupted database in the first place. It would be nice for somebody to verify that the update works for them.

File: pkgs/servers/mail/mailman/web.nix [hunk 1]
```diff
@@ -15,6 +15,13 @@ buildPythonPackage rec {
     leaveDotGit = true;
   };
 
+  patches = [
+    (fetchpatch {
```
Comment (jonringer) [inline pkgs/servers/mail/mailman/web.nix:19]: do you mind adding a comment about what the intent of the patch is doing, and when the patch is likely not to be needed
```suggestion
  patches = [
    # does x, y, z. Should be able to be removed after >=X.Y.Z
    (fetchpatch {
```

**Comment by benley** [comment]
> I think @benley has one.

Yes, but I haven't had time to work on that project for a bit.  If I can get back into it soon I'll test this.

**Comment by alyssais** [comment]
@lheckemann are you able to test this?

**Comment by lheckemann** [comment]
I have some general rework on the module in progress, this is part of it so I'm definitely for this change — though as I mentioned upstream maybe we should remove the django dependency (since it's a transitive dependency) rather than loosening the version bounds? That said, I'm fine with merging this with an explanatory comment as @jonringer suggested (and I'm already using this effectively)

**Comment by lheckemann** [comment]
https://github.com/mayflower/nixpkgs/commit/477ebb731ac6cd42537105d910d887b0a4f8fafe I'd include this in the module PR and we could supersede this one with it?

**Comment by FRidh** [comment]
@lheckemann if I am correct, now that https://github.com/NixOS/nixpkgs/pull/86177 is in, this can be closed?

**Comment by lheckemann** [comment]
Yes.

**Pull request closed without merging**